src/mixxxapplication.cpp

```diff
@@ -207,10 +217,43 @@ bool MixxxApplication::notify(QObject* target, QEvent* event) {
 #endif // QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
 #endif // Q_OS_MAC
 
+bool MixxxApplication::notify(QObject* target, QEvent* event) {
+    // All touch events are translated in two simultaneous events one for
+    // target QWidgetWindow and one for the target QWidget
+    // A second touch becomes a mouse move without additional press and release
+    // events
+    switch (event->type()) {
+    case QEvent::MouseButtonPress: {
+        QMouseEventEditable* mouseEvent = static_cast<QMouseEventEditable*>(event);
+        if (mouseEvent->source() == Qt::MouseEventSynthesizedByQt &&
+                mouseEvent->button() == Qt::LeftButton &&
+                touchIsRightButton()) {
+            mouseEvent->setButton(Qt::RightButton);
+            m_rightPessedButtons++;
```

Comment:
VERIFY_OR_DEBUG_ASSERT(m_rightPessedButtons < 2) and early break otherwise?

Reply:
Solved, see above.